qa-tests-backend/src/main/groovy/util/Helpers.groovy

```diff
@@ -65,6 +65,24 @@ class Helpers {
         return willRetry
     }
 
+    static <V> V waitForTrue(int retries, int intervalSeconds, Closure <V> closure) {
+        Timer t = new Timer(retries, intervalSeconds)
+        int attempt = 0
+        boolean result = false
+        while (t.IsValid()) {
+            attempt++
+            result = closure()
+            if (result) {
+                return true
+            }
+            log.debug "Attempt ${attempt} failed, retrying"
+        }
+        if (!result) {
+            throw new RuntimeException("All ${attempt} attempts failed, could not reach desired state")
+        }
+        return result
```

Comment:
nit: will never return so just throw.
```suggestion
        throw new RuntimeException("All ${attempt} attempts failed, could not reach desired state")
```

Reply:
Good point, when the timer expires then `result` must be `false` :)

Reply:
I patched that and also removed the `result` var as it was no longer needed. I also changed the template `<V>` to `boolean` as IDE suggested.